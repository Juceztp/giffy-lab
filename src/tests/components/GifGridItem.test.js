import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Prueba del componente GifGridItem', () => {

    const gif = {
        title: 'naruto shippuden GIF',
        url: 'https://media3.giphy.com/media/4gsjHZMPXdlGo/giphy.gif?cid=334fdb12wgy0f90xni318my9wi58q34pggrthsczwekvl0si&amp;rid=giphy.gif'
    };
    let wrapper = shallow( <GifGridItem {...gif} /> );

    beforeEach(() => {
        wrapper = shallow( <GifGridItem {...gif} /> );
    });

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un pÃ¡rrafo con el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(gif.title);
    });

    test('Debe de tener la imagen igual a la url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(gif.url);
        expect(img.prop('alt')).toBe(gif.title);
    });
    
    test('Debe de tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        const classes = div.prop('className');
        expect(classes.includes('animate__fadeIn')).toBe(true);
    });
    
});