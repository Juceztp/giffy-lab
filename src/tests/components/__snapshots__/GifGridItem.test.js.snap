// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card-gif animate__animated animate__fadeIn"
>
  <img
    alt="naruto shippuden GIF"
    src="https://media3.giphy.com/media/4gsjHZMPXdlGo/giphy.gif?cid=334fdb12wgy0f90xni318my9wi58q34pggrthsczwekvl0si&amp;rid=giphy.gif"
  />
  <p>
    naruto shippuden GIF
  </p>
</div>
`;
